File: backend/src/main/java/org/gestionstock/stock/Payload/Response/UserResponse.java
package org.gestionstock.stock.Payload.Response;

import java.util.Set;

import org.gestionstock.stock.Enum.ERole;

public record UserResponse(
    String id,
    String username,
    String email,
    String firstName,
    String lastName,
    String address,
    String phone,
    String imageUrl,
    Set<ERole> roles,
    String createdAt,
    String updatedAt
) {
    
}

File: frontend/src/app/models/user-response.ts
import { ERole } from "./erole";

export interface UserResponse {
    id: string;
    username: string;
    email: string;
    firstName: string;
    lastName: string;
    address: string;
    phone: string;
    imageUrl?: string;
    roles: ERole[];
    createdAt: Date;
    updatedAt: Date;
}
